File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.js
function solution(n,a,b)
{
    var answer = 1;

    while(true) {
        
        if(Math.abs(a-b) === 1 && Math.min(a,b) %2 === 1) break;
        
        if(a%2 === 0) a/=2;
        else a= Math.floor(a/2)+ 1;
        
        if(b%2 === 0) b/=2;
        else b= Math.floor(b/2)+ 1;
        
        answer +=1;
    } 

    return answer;
}

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.cpp
#include <iostream>
#include <cmath>

using namespace std;

int solution(int n, int a, int b)
{
    int answer = 0;

    while(a != b) {
        a = ceil(float(a)/2);
        b = ceil(float(b)/2);
        
        answer++;
        
    }

    return answer;
}